gammapy/datasets/tests/test_simulate.py

```diff
@@ -126,22 +159,61 @@ def dataset():
 
 
 @requires_data()
-def test_sample_coord_time_energy(dataset, models):
-    models[0].spatial_model = None
-    dataset.models = models
+def test_evaluate_timevar_source(enedip_temporal_model, dataset):
+    dataset.models = enedip_temporal_model
+    evaluator = dataset.evaluators["test-source"]
+
+    sampler = MapDatasetEventSampler(random_state=0)
+    npred = sampler._evaluate_timevar_source(dataset, evaluator)
+
+    assert_allclose(np.shape(npred.data), (3, 1999, 1, 1))
+
+    assert_allclose(npred.data[:, 10, 0, 0], [0.024827, 0.110424, 0.321845], rtol=2e-4)
+    assert_allclose(npred.data[:, 50, 0, 0], [0.024827, 0.110424, 0.321845], rtol=2e-4)
+
+    filename = "$GAMMAPY_DATA/gravitational_waves/GW_example_DC_map_file.fits.gz"
+    temporal_model = LightCurveTemplateTemporalModel.read(filename, format="map")
+    temporal_model.t_ref.value = 51544.00074287037
+    dataset.models[0].temporal_model = temporal_model
+    evaluator = dataset.evaluators["test-source"]
+
+    sampler = MapDatasetEventSampler(random_state=0)
+    npred = sampler._evaluate_timevar_source(dataset, evaluator)
+
+    assert_allclose(
+        npred.data[:, 1000, 0, 0] / 1e-13,
+        [0.038113, 0.033879, 0.010938],
+        rtol=2e-4,
+    )
+
+
+@requires_data()
+def test_sample_coord_time_energy(dataset, enedip_temporal_model):
+    enedip_temporal_model.spatial_model = None
```

Comment:
The sampling cannot work if there is no spatial model (ie, it does not assume a point spatial model?)

Reply:
It works only for point-like morphologies. It does not assume a point-like one if the `spatial_model` is `None`: it only raises an error! 
I've also added a test when the morphology is Gaussian.

Reply:
Does event sampling work for point like IRFs?